src/services/auth-service.ts

```diff
@@ -0,0 +1,128 @@
+import { repository } from '@loopback/repository';
+import { JWTService } from './jwt-service';
+import { securityId, UserProfile } from '@loopback/security';
+import { TokenService } from '@loopback/authentication';
+import { TokenServiceBindings } from '@loopback/authentication-jwt';
+import { inject } from '@loopback/core';
+import winston from 'winston';
+import { LogConfig } from '../config/logConfig';
+import { HttpErrors } from '@loopback/rest';
+import { TokenKvRepository } from '../repositories';
+import { TokenKv } from '../models';
+
+
+
+export interface Tokens {
+  accessToken: string,
+  refreshToken: string,
+  expiresIn: number
+}
+
+export class AuthService {
+  constructor(
+    @repository(TokenKvRepository)
+    public redisRepo: TokenKvRepository,
+    @inject(TokenServiceBindings.TOKEN_SERVICE)
+    public jwtService: TokenService,
+    public logger = winston.loggers.get(LogConfig.logName),
+  ) { }
+
+  /**
+   * Takes a users and returns access and refresh tokens. 
+   * It also saves tokens into redis with ttl of 2 hours
+   * @param userP User Profile
+   * @returns 
+   */
+  async createTokens(userP: UserProfile): Promise<Tokens> {
+    let accToken = '';
+    let refToken = '';
+
+    const userId = String(userP[securityId])
```

Comment:
any reason to use `String` class for coercion here?

Reply:
There is no particular reason actually, `userP[securityId].toString()` will also work fine, just that `toString()` will through an error when `userId` is null and `String` would not. But in this case, `userId` will not be null so it will be okay to use that as well.